migrations/m150623_212711_fix_username_notnull.php

```diff
@@ -36,7 +36,7 @@ public function down()
             if ($this->dbType == 'sqlsrv') {
                 $this->dropIndex('{{%user_unique_username}}', '{{%user}}');
             }
-            $this->alterColumn('{{%user}}', 'username', $this->string(255)->null());
+            $this->alterColumn('{{%user}}', 'username', $this->string(255), ' NULL');
```

Comment:
@dmeroff ok should be fixed now. I tested all up and down on SQL Server 2012 and it works correctly.